src/test/java/com/google/sps/MatchServletTest.java

```diff
@@ -102,6 +103,26 @@ public void nullParameterValues_ShouldNotThrowError() throws IOException, Servle
         verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
     }
     
+    @Test
+    public void nullUser_ShouldReturnNull() throws IOException, ServletException {
```

Comment:
Could you also test the case where a logged out user makes a post request to the servlet?

Reply:
Done